lib/fastlane/plugin/wpmreleasetoolkit/helper/android/android_version_helper.rb

```diff
@@ -30,42 +30,77 @@ module VersionHelper
+        # @return [Hash] A hash with 2 keys "name" and "code" containing the extracted version name and code, respectively
+        #
+        def self.get_release_version(app)
+          if properties_file_exists
+            section = ENV['HAS_ALPHA_VERSION'].nil? ? 'defaultConfig' : 'vanilla {'
+            gradle_path = self.gradle_path
+            name = get_version_name_from_gradle_file(gradle_path, section)
+            code = get_version_build_from_gradle_file(gradle_path, section)
+            return { VERSION_NAME => name, VERSION_CODE => code }
+          end
+
+          return get_version_from_properties(product_name: app)
+        end
+
+        def self.properties_file_exists
+          properties_file_path = File.join(ENV['PROJECT_ROOT_FOLDER'] || '.', 'version.properties')
+
+          return File.exist?(properties_file_path)
+        end
+
+        # Extract the version name and code from the `version.properties` file in the project root
         #
         # @return [Hash] A hash with 2 keys "name" and "code" containing the extracted version name and code, respectively
         #
-        def self.get_release_version
-          section = ENV['HAS_ALPHA_VERSION'].nil? ? 'defaultConfig' : 'vanilla {'
-          gradle_path = self.gradle_path
-          name = get_version_name_from_gradle_file(gradle_path, section)
-          code = get_version_build_from_gradle_file(gradle_path, section)
-          return { VERSION_NAME => name, VERSION_CODE => code }
+        def self.get_version_from_properties(product_name:, is_alpha: false)
+          version_name_key = "#{product_name}.#{is_alpha ? 'alpha.' : ''}versionName"
+          version_code_key = "#{product_name}.#{is_alpha ? 'alpha.' : ''}versionCode"
+
+          properties_file_path = File.join(ENV['PROJECT_ROOT_FOLDER'] || '.', 'version.properties')
+
+          return nil unless File.exist?(properties_file_path)
+
+          File.open(properties_file_path, 'r') do |f|
+            text = f.read
+            name = text.match(/#{version_name_key}=(\S*)/m)&.captures&.first
+            code = text.match(/#{version_code_key}=(\S*)/m)&.captures&.first
+
+            f.close
+
+            return nil if name.nil? || code.nil?
+
+            return { VERSION_NAME => name, VERSION_CODE => code.to_i }
+          end
         end
 
-        # Extract the version name and code from the `defaultConfig` of the `$PROJECT_NAME/build.gradle` file
+        # Extract the version name and code from the `version.properties` file in the project root
         #
         # @return [Hash] A hash with 2 keys `"name"` and `"code"` containing the extracted version name and code, respectively,
         #                or `nil` if `$HAS_ALPHA_VERSION` is not defined.
         #
-        def self.get_alpha_version
-          return nil if ENV['HAS_ALPHA_VERSION'].nil?
+        def self.get_alpha_version(app)
+          if properties_file_exists
```

Comment:
Same here, `if properties_file_exists` is true we call `get_version_name_from_gradle_file` et al…

Reply:
Ahhhh good catch!